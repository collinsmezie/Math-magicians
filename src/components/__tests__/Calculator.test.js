import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import CalculatorInterface from '../Calculator';

describe('Calculator page of the App', () => {
  test('Builds the snapshot of the Calculator component safely', () => {
    const rend = renderer.create(<CalculatorInterface />);
    expect(rend.toJSON()).toMatchSnapshot();
  });
  test('Render Calculator component', () => {
    render(<CalculatorInterface />);
  });
});
