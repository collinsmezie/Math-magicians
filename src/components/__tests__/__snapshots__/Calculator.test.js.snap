// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page of the App Builds the snapshot of the Calculator component safely 1`] = `
Array [
  <div
    className="text"
  >
    <h2>
      Let's Do some Math!
    </h2>
  </div>,
  <div
    className="container"
  >
    <div
      className="display"
    >
      <h1 />
    </div>
    <div
      className="btns"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        className="button zero"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>,
]
`;
